validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorConfig.java

```diff
@@ -58,17 +60,23 @@ private ValidatorConfig(
     this.validatorExternalSignerUrl = validatorExternalSignerUrl;
     this.validatorExternalSignerTimeout = validatorExternalSignerTimeout;
     this.graffiti = graffiti;
-    this.validatorPerformanceTrackingEnabled = validatorPerformanceTrackingEnabled;
     this.validatorKeystoreLockingEnabled = validatorKeystoreLockingEnabled;
     this.beaconNodeApiEndpoint = beaconNodeApiEndpoint;
+    if (validatorPerformanceTrackingEnabled == null) {
+      this.validatorPerformanceTrackingMode = validatorPerformanceTrackingMode;
+    } else if (validatorPerformanceTrackingEnabled) {
+      this.validatorPerformanceTrackingMode = ValidatorPerformanceTrackingMode.ALL;
+    } else {
+      this.validatorPerformanceTrackingMode = ValidatorPerformanceTrackingMode.NONE;
+    }
```

Comment:
I think this should probably just be done in `ValidatorOptions.configure` and this class would only take a `ValidatorPerformanceTrackingMode` so the backwards compatibility is isolated.

Reply:
Done.